spec/commands/update_spec.rb

```diff
@@ -635,6 +635,26 @@
     end
   end
 
+  context 'with --source option' do
+    before do
+      build_repo2 do
+        build_gem "rack", "2.0"
+      end
+    end
+
+    it 'updates gems using the given source' do
+      gemfile <<-G
+        source "file://#{gem_repo1}"
+        gem 'rack'
+      G
+
+      bundle "install"
+
+      bundle "update --source file://#{gem_repo2}"
+      expect(out).to include "Fetching source index from file:#{gem_repo2}/"
```

Comment:
I realise a better test for this would be to actually check `the_bundle` instead